### ISSUE TITLE
Precinct lacks support for TypeScript's template literal types

### ISSUE THREAD

**Issue by casr** [dependency-check-team/dependency-check#226]
I think this is likely an upstream problem with Precinct but just thought you'd like to be aware that this error might occur:

```
unexpected error: TSError: Type expected.
    at createError (~/.npm/_npx/b799cd4caceb105d/node_modules/@typescript-eslint/typescript-estree/dist/node-utils.js:524:12)
    at convertError (~/.npm/_npx/b799cd4caceb105d/node_modules/@typescript-eslint/typescript-estree/dist/convert.js:36:41)
    at astConverter (~/.npm/_npx/b799cd4caceb105d/node_modules/@typescript-eslint/typescript-estree/dist/ast-converter.js:15:42)
    at parseWithNodeMapsInternal (~/.npm/_npx/b799cd4caceb105d/node_modules/@typescript-eslint/typescript-estree/dist/parser.js:403:66)
    at Object.parse (~/.npm/_npx/b799cd4caceb105d/node_modules/@typescript-eslint/typescript-estree/dist/parser.js:361:21)
    at module.exports.parse (~/.npm/_npx/b799cd4caceb105d/node_modules/node-source-walk/index.js:57:22)
    at module.exports.walk (~/.npm/_npx/b799cd4caceb105d/node_modules/node-source-walk/index.js:99:52)
    at module.exports (~/.npm/_npx/b799cd4caceb105d/node_modules/detective-typescript/index.js:33:10)
    at precinct (~/.npm/_npx/b799cd4caceb105d/node_modules/precinct/index.js:111:20)
    at file://~/.npm/_npx/b799cd4caceb105d/node_modules/dependency-check/lib/extensions.js:60:29 {
  fileName: '~/Projects/test/website/estree.ts',
  index: 2616,
  lineNumber: 96,
  column: 57
}
```

with code that resembles:

```typescript
type X1 = `test_${string}`
```
